lib/AST/ASTImporter.cpp

```diff
@@ -2205,8 +2216,17 @@ Decl *ASTNodeImporter::VisitRecordDecl(RecordDecl *D) {
             // The record types structurally match, or the "from" translation
             // unit only had a forward declaration anyway; call it the same
             // function.
-            // FIXME: For C++, we should also merge methods here.
-            return Importer.MapImported(D, FoundDef);
+            // FIXME: Check for ODR.
```

Comment:
Why do we need this FIXME? Didn't we check already for ODR? There is a structural match at this point, so we know exactly that this is a similar definition with the exact same field members.

Reply:
Methods can be different in the two record definitions (theoretically).

Reply:
The ODR check of the methods is happening when we call the import on that method (structural eq check does that), so I still don't think this `FIXME` is valid.

Reply:
The bad case can be if there is a class:
```
struct X { void f1(); };
```
And such is imported:
```
struct X { void f2(int); };
```
If not the merge (X will contain f1 and f2) is the correct way of importing this there could be a warning for this case. (If the ASTImporter is supposed to work only with source code of a single valid program probably no such case happens.)

Reply:
Probably the FIXME can be changed to "check for same methods".